File: source/GyeongHo/boj/10845.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
q = deque()
for _ in range(n):
    command = input().rstrip()
    if command == 'pop':
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif command == 'size':
        print(len(q))
    elif command == 'empty':
        if q:
            print(0)
        else:
            print(1)
    elif command == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    elif command == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)
    else:
        q.append(int(command.split()[1]))

File: source/GyeongHo/boj/10845.cpp
#include <bits/stdc++.h>
using namespace

class q
{
private:
	int arr[10001];
	int head = 0;
	int tail = 0;

public:
	int push(int num)
	{
		if (tail + 1 > 10001)
			return 0;
		arr[tail++] = num;
		return 1;
	}

	void pop()
	{
		if (++head > 10000)
			return 0;
		return 1;
	}

	int size()
	{
		return tail - head;
	}

	int empty()
	{
		if (tail == 0)
			return 1;
		return 0;
	}

	int front()
	{
		if (tail == 0)
			return -1;
		return arr[head];
	}

	int back()
	{
		if (tail == 0)
			return -1;
		return arr[tail - 1];

int main()
{
	q q1;
